types/index.d.ts

```diff
@@ -38,7 +38,7 @@ declare namespace visit {
   type Visitor<V extends Node> = (
     node: V,
     index: number,
-    parent: Node
+    parent: Parent
```

Comment:
Root elements have no `parent`, in that case this value would be `undefined`
```suggestion
    parent: Parent | undefined
```